### PR TITLE
Zendesk Mobile: add support for source tags

### LINKED ISSUE #9290: Zendesk Mobile: add support for SupportSourceTag
The Helpshift `SupportViewController` has the following SupportSourceTags. Need to support them for Zendesk, and add use the source as the 'origin' tag on Zendesk tickets.

```
SupportSourceTag const SupportSourceTagWPComLogin = @"origin:wpcom-login-screen";
SupportSourceTag const SupportSourceTagWPComSignup = @"origin:signup-screen";
SupportSourceTag const SupportSourceTagWPOrgLogin = @"origin:wporg-login-screen";
SupportSourceTag const SupportSourceTagJetpackLogin = @"origin:jetpack-login-screen";
SupportSourceTag const SupportSourceTagGeneralLogin = @"origin:login-screen";
SupportSourceTag const SupportSourceTagInAppFeedback = @"origin:in-app-feedback";
SupportSourceTag const SupportSourceTagLoginEmail = @"origin:login-email";
SupportSourceTag const SupportSourceTagLoginMagicLink = @"origin:login-magic-link";
SupportSourceTag const SupportSourceTagSignupMagicLink = @"origin:signup-magic-link";
SupportSourceTag const SupportSourceTagLoginWPComPassword = @"origin:login-wpcom-password";
SupportSourceTag const SupportSourceTagLogin2FA = @"origin:login-2fa";
SupportSourceTag const SupportSourceTagLoginSiteAddress = @"origin:login-site-address";
SupportSourceTag const SupportSourceTagLoginUsernamePassword = @"origin:login-username-password";
SupportSourceTag const SupportSourceTagWPComCreateSiteCategory = @"origin:wpcom-create-site-category";
SupportSourceTag const SupportSourceTagWPComCreateSiteTheme = @"origin:wpcom-create-site-theme";
SupportSourceTag const SupportSourceTagWPComCreateSiteDetails = @"origin:wpcom-create-site-details";
SupportSourceTag const SupportSourceTagWPComCreateSiteDomain = @"origin:wpcom-create-site-domain";
SupportSourceTag const SupportSourceTagWPComCreateSiteUsername = @"origin:wpcom-create-site-username";
SupportSourceTag const SupportSourceTagWPComCreateSiteCreation = @"origin:wpcom-create-site-creation";
SupportSourceTag const SupportSourceTagWPComSignupEmail = @"origin:wpcom-signup-email-entry";
SupportSourceTag const SupportSourceTagSignupWaitingForGoogle = @"origin:signup-waiting-for-google";
```

### PR DESCRIPTION
**Pull request by ScoutHarris**
Fixes #9290 

**NOTE:** 
Zendesk currently does not work if you are not logged in. If you want to test login/signup tags, in `ZendeskUtils:createIdentity`, change the `blog` guard to this:
```
guard let blog = blogService.lastUsedBlog() else {
    DDLogInfo("No Blog to create Zendesk identity with.")
    ZendeskUtils.sharedInstance.userEmail = "anyemailaddress@gmail.com"
    ZendeskUtils.createZendeskIdentity()
    completion(true)
    return
}
```

**To test:**
- In the app, access Support from a location that has a source tag. The list can be found in `WordPressSupportSourceTag+Helpers.swift`. 
  - The easiest/quickest may be the Site Creation flow > navigation bar 'Help'. 
- Select 'Contact Us' from the Support options.

To verify the origin tag is added to the ticket, you can either:
- Go ahead and submit a ticket.
- Find your ticket in Zendesk and verify the origin tag is present. (Feel free to ask me to look at your ticket. Might be easier.)

**OR**

- In `ZendeskUtils:getTags` log out the `tags` array before the method returns.
- When you select 'Contact Us' `tags` will appear in the logs.
- Verify the origin tag is present. 

@jleandroperez - can I bother you for a review please? Thank you!!

### CODE REVIEW

File: WordPressAuthenticator/WordPressAuthenticator/Authenticator/WordPressSupportSourceTag.swift [hunk 1]
```diff
@@ -13,11 +13,16 @@ public enum WordPressSupportSourceTag {
...
     case wpComCreateSiteDetails
     case wpComCreateSiteUsername
     case wpComSignupEmail
     case wpComSignup
     case wpComSignupWaitingForGoogle
     case wpComSignupMagicLink
+    case wpComLogin
+    case wpOrgLogin
+    case inAppFeedback
 }
```
Comment (jleandroperez) [inline WordPressAuthenticator/WordPressAuthenticator/Authenticator/WordPressSupportSourceTag.swift]: This one is potentially problematic, long term. Reasons:

- This class lives in a framework that does not have knowledge of the `Create Sites` flow.
- inAppFeedback is also Client-App Side.

Suggestion!!. I think it's time we converted `WordPressSupportSourceTag` into an extensible enum. This way we get to keep the "Enum Cases" we need, in the authenticator, and the client app can piggyback, if needed.

Will send you a PR in a little bit: #9315

Huge WIN about this approach is that: when the Authenticator is in it's own repository, we'd be able to just add "new WordPressSupportSourceTag cases" without the need to actually edit this file. (Details coming up in that PR).

File: WordPress/Classes/Utility/ZendeskUtils.swift [hunk 2]
```diff
@@ -39,13 +45,15 @@ import CoreTelephony
         ZendeskUtils.toggleZendesk(enabled: true)
     }
 
-    func showHelpCenterIfPossible(from controller: UIViewController) {
+    func showHelpCenterIfPossible(from controller: UIViewController, withSourceTag sourceTag: WordPressSupportSourceTag? = nil) {
```
Comment (jleandroperez) [inline WordPress/Classes/Utility/ZendeskUtils.swift]: Suggestion!

```
func showHelpCenterIfPossible(from controller: UIViewController, with sourceTag: WordPressSupportSourceTag? = nil)
```

(OR)

```
func showHelpCenterIfPossible(from controller: UIViewController, sourceTag: WordPressSupportSourceTag? = nil)
```

(The first one is probably more Swifty!)

File: WordPress/Classes/Utility/ZendeskUtils.swift [hunk 3]
```diff
@@ -60,27 +68,31 @@ import CoreTelephony
         }
     }
 
-    func showNewRequestIfPossible(from controller: UIViewController) {
+    func showNewRequestIfPossible(from controller: UIViewController, withSourceTag sourceTag: WordPressSupportSourceTag? = nil) {
```
Comment (jleandroperez) [inline WordPress/Classes/Utility/ZendeskUtils.swift]: Same naming suggestion(s) as above!. How about `with sourceTag:` (OR) `sourceTag:`

File: WordPress/Classes/Utility/WPError.m [hunk 4]
```diff
@@ -178,7 +178,7 @@ + (void)showAlertWithTitle:(NSString *)title message:(NSString *)message withSup
                                                                
                                                                if ([Feature enabled:FeatureFlagZendeskMobile]) {
                                                                    SupportTableViewController *supportVC = [SupportTableViewController new];
-                                                                   supportVC.sourceTag = sourceTag;
+                                                                   supportVC.sourceTagDescription = sourceTag;
```
Comment (jleandroperez) [inline WordPress/Classes/Utility/WPError.m]: This was actually mapping the `sourceTag` (String) into a Swift Enum Case. Did you have any issues with this mapping?

If we could revert this change, `SupportVC` could just have `sourceTag` (we could remove `sourceTagDescription`, and simplify a little bit everything!).

Comment (ScoutHarris) [reply WordPress/Classes/Utility/WPError.m]: Just to recap what we discussed, `WordPressSupportSourceTag` is a Swift enum that can't be used in ObjC. So I still need `sourceTagDescription`,  but I will "hide the dirt". 😄

**Comment by ScoutHarris** [comment]
OK @jleandroperez let's see if you like this version better. 😉 Ready for another review.
